### PR TITLE
Use Conan for installing dependencies

### LINKED ISSUE #295: Use conan to install dependencies
https://conan.io/

### LINKED ISSUE #306: Update README on building with sopt conan package

### LINKED ISSUE #309: Update algo factory with sopt improvements
After https://github.com/astro-informatics/sopt/pull/305 is merged to sopt, update algo factory to create the ForwardBackward objects using injection of g_proximal.

### LINKED ISSUE #310: Workflow dispatch not working on cmake workflow
We have the `workflow_dispatch` in the `cmake.yml` workflow to use Tmate, but it doesn't seem to show up as runnable on the Actions page.

### LINKED ISSUE #313: Algo_factory test with ForwardBackward failing
We get a segfault from `is_converged()` function. The problem appears to be happening inside the g_proximal proximal_norm within the objective_convergence function

```
Thread 1 "test_algo_facto" received signal SIGSEGV, Segmentation fault.
0x000055555562c3b8 in Eigen::DenseStorage<std::complex<double>, -1, -1, 1, 0>::rows (this=0x0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/DenseStorage.h:544
544         EIGEN_DEVICE_FUNC Index rows(void) const {return m_rows;}
(gdb) bt
#0  0x000055555562c3b8 in Eigen::DenseStorage<std::complex<double>, -1, -1, 1, 0>::rows (this=0x0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/DenseStorage.h:544
#1  0x000055555562399a in Eigen::PlainObjectBase<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> >::rows (this=0x0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/PlainObjectBase.h:151
#2  0x00005555556846d5 in Eigen::ArrayWrapper<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const>::rows (this=0x7fffffffc5f0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/ArrayWrapper.h:64
#3  0x00005555556ca5be in Eigen::EigenBase<Eigen::ArrayWrapper<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const> >::rows (this=0x7fffffffc5f0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/EigenBase.h:59
#4  0x00005555556c6c91 in Eigen::EigenBase<Eigen::ArrayWrapper<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const> >::size (this=0x7fffffffc5f0)
    at /home/mmcleod/.conan/data/eigen/3.3.7/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eigen3/Eigen/src/Core/EigenBase.h:66
#5  0x00005555556c5af3 in sopt::l1_norm<Eigen::ArrayWrapper<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const>, Eigen::ArrayWrapper<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const> > (input=..., 
    weights=...) at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/maths.h:118
#6  0x00005555556c49f7 in sopt::l1_norm<Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1>, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> > (input=..., weight=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/maths.h:125
#7  0x00005555556c3776 in sopt::algorithm::L1GProximal<std::complex<double> >::proximal_norm (this=0x55555671ebe0, x=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/l1_g_proximal.h:52
#8  0x0000555555643159 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::objective_convergence (this=0x5555565d84b0, obj_comm=..., scalvar=..., x=..., residual=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:319
#9  0x0000555555634a42 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::is_converged (this=0x5555565d84b0, scalvar=..., x=..., residual=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:331
#10 0x0000555555629e13 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::operator()(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1>&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&) const::{lambda(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&)#2}::operator()(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&) (this=0x5555565d84b0, x=..., residual=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:271
--Type <RET> for more, q to quit, c to continue without paging--
#11 0x000055555564366c in std::_Function_handler<bool (Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&), sopt::algorithm::ImagingForwardBackward<std::complex<double> >::operator()(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1>&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&) const::{lambda(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&)#2}>::_M_invoke(std::_Any_data const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&) (__functor=..., __args#0=..., __args#1=...) at /usr/include/c++/9/bits/std_function.h:285
#12 0x0000555555642e77 in std::function<bool (Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&)>::operator()(Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&, Eigen::Matrix<std::complex<double>, -1, 1, 0, -1, 1> const&) const (this=0x7fffffffcb80, __args#0=..., __args#1=...) at /usr/include/c++/9/bits/std_function.h:688
#13 0x000055555565187a in sopt::algorithm::ForwardBackward<std::complex<double> >::is_converged (this=0x7fffffffcb50, x=..., residual=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/forward_backward.h:132
#14 0x0000555555643e03 in sopt::algorithm::ForwardBackward<std::complex<double> >::operator() (this=0x7fffffffcb50, out=..., x_guess=..., res_guess=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/forward_backward.h:274
#15 0x00005555556351b7 in sopt::algorithm::ForwardBackward<std::complex<double> >::operator() (this=0x7fffffffcb50, out=..., guess=std::tuple containing = {...})
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/forward_backward.h:149
#16 0x000055555562a3a3 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::operator() (this=0x5555565d84b0, out=..., guess=..., res=...)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:283
#17 0x00005555556224a7 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::operator() (this=0x5555565d84b0, out=..., guess=std::tuple containing = {...})
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:161
#18 0x000055555561b518 in sopt::algorithm::ImagingForwardBackward<std::complex<double> >::operator() (this=0x5555565d84b0)
    at /home/mmcleod/.conan/data/sopt/4.0.0/_/_/package/bae3b45c509735da37c9f0e38d6e5a1de907180c/include/sopt/imaging_forward_backward.h:187
#19 0x00005555555fe7f8 in C_A_T_C_H_T_E_S_T_0 () at /home/mmcleod/lexci/purify/cpp/tests/algo_factory.cc:172
#20 0x00005555556f7124 in Catch::TestInvokerAsFunction::invoke (this=0x5555565d33f0) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:14332
#21 0x00005555556f6361 in Catch::TestCase::invoke (this=0x5555565fa8f0) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:14171
#22 0x00005555556f08d6 in Catch::RunContext::invokeActiveTestCase (this=0x7fffffffd7c0) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:13027
#23 0x00005555556f05fa in Catch::RunContext::runCurrentTest (this=0x7fffffffd7c0, redirectedCout="", redirectedCerr="")
    at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:13000
--Type <RET> for more, q to quit, c to continue without paging--
#24 0x00005555556ef0bc in Catch::RunContext::runTest (this=0x7fffffffd7c0, testCase=...) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:12761
#25 0x00005555556f21b5 in Catch::(anonymous namespace)::TestGroup::execute (this=0x7fffffffd7b0)
    at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:13354
#26 0x00005555556f3577 in Catch::Session::runInternal (this=0x7fffffffda60) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:13564
#27 0x00005555556f326d in Catch::Session::run (this=0x7fffffffda60) at /home/mmcleod/.conan/data/catch2/2.13.9/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/catch2/catch.hpp:13520
#28 0x00005555557098a1 in main (argc=1, argv=0x7fffffffdca8) at /home/mmcleod/lexci/purify/cpp/tests/common_catch_main.cc:24
```

### LINKED ISSUE #314: Update README
The README is out of date in a few places including mentioning [Travis CI](https://github.com/astro-informatics/purify/blob/5833dbd35f214212aa17c368984bcae180cf5dee/README.md?plain=1#L20)

### LINKED ISSUE #316: CI failing on Ubuntu because of ccache timestamp
GitHub Actions are failing on Ubuntu in the [prepare ccache timestamp](https://github.com/astro-informatics/purify/actions/runs/5269851162/workflow?pr=315#L119) step

Output:
```
Run echo "{date_and_time}={$(date +'%Y-%m-%d-%H;%M;%S')}" >> $GITHUB_OUTPUT
  echo "{date_and_time}={$(date +'%Y-%m-%d-%H;%M;%S')}" >> $GITHUB_OUTPUT
  shell: /usr/local/bin/cmake -P {0}
  env:
    BUILD_TYPE: Release
    OMP_NUM_THREADS: 2
    CONAN_PRINT_RUN_COMMANDS: 1
    CONAN_CPU_COUNT: 2
    CONAN_SKIP_BROKEN_SYMLINKS_CHECK: True
    CC: gcc-9
    CXX: g++-9
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.11/x64/lib:/usr/local/lib
    pythonLocation: /opt/hostedtoolcache/Python/3.10.11/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.10.11/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
CMake Error at /home/runner/work/_temp/f09b53e7-0e9b-4d9e-9b8d-b4e905aba3dc:1:
  Parse error.  Expected "(", got quoted argument with text
  "{date_and_time}={$(date +'%Y-%m-%d-%H;%M;%S')}".


CMake Error: Error processing file: /home/runner/work/_temp/f09b53e7-0e9b-4d9e-9b8d-b4e905aba3dc
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by SJaffa**
Starting from the development branch, first issue was the same spdlog error as in [sopt](https://github.com/astro-informatics/sopt/pull/265/files)

- [x] Update install instructions in README.md with https://github.com/astro-informatics/purify/pull/302#issuecomment-1532985806
- [x] Fix code errors in algo_factory.h
- [x] Only include sopt TF headers from algo_factory.h when `cppflow==on` option is passed to cmake
- [x] In sopt, move `class GProximal` to the `sopt::algorithm` namespace
- [ ] Update documentation GitHub Actions workflow

### CODE REVIEW

**Comment by SJaffa** [comment]
- Checks fail
	- All Ubuntu fails in build step because of spdlog error in SOPT, need to pull SOPT to fix this
	- macOS clang also fails in build step because of spdlog error
	- macOS gcc fails in 'install conan on gcc' 
	    - boost fiber error:

`boost/1.78.0: WARN: Boost component 'fiber' is missing libraries. Try building boost with '-o boost:without_fiber'. (Option is not guaranteed to exist)
ERROR: boost/1.78.0: Error in package_info() method, line 1592
	raise ConanException("These libraries were expected to be built, but were not built: {}".format(non_built))
	ConanException: These libraries were expected to be built, but were not built: {'boost_fiber'}`

**Comment by tkoskela** [comment]
We need to close https://github.com/astro-informatics/sopt/issues/279 before we can make progress with this

**Comment by SJaffa** [comment]
@tkoskela is this the branch you got the SOPT package working on? This one fails for me as it can't find Eigen and spdlog.

**Comment by tkoskela** [comment]
The algo_factory that creates the `ImagingForwardBackward` object in purify is templated by the algorithm type https://github.com/astro-informatics/purify/blob/5833dbd35f214212aa17c368984bcae180cf5dee/cpp/main.cc#L440 but the template parameter is checked by an if statement and there's a different branch for each algorithm class https://github.com/astro-informatics/purify/blob/5833dbd35f214212aa17c368984bcae180cf5dee/cpp/purify/algorithm_factory.h#L142

:zany_face:

**Comment by SJaffa** [comment]
The Cmake flags for building SOPT are specified in [cmakefiles/LookUpSopt.cmake](https://github.com/astro-informatics/purify/blob/0a940e409e95bfbddf4a9d74daa9daad56938165/cmake_files/LookUpSopt.cmake#L34) so we can add an option in the `conanfile.py` to turn `cppflow` on and off and pass that through to SOPT like they do with `openmp` and `logging`

**Comment by tkoskela** [comment]
TODO: Add these instructions to README

To build `purify` using this PR and https://github.com/astro-informatics/sopt/pull/350

- [Install tensorflow C api](https://serizba.github.io/cppflow/installation.html)
- git clone the UCL fork of cppflow and create a conan package using
```
git clone git@github.com:UCL/cppflow.git
conan create ./cppflow/ -pr:h=default -pr:b=default
```
- git clone sopt and create a conan package using 
```
git clone git@github.com:astro-informatics/sopt.git
conan create ./sopt/ --build missing -s compiler.libcxx=libstdc++11 -o cppflow=on  -pr:h=default -pr:b=default
```
- conan install and build as usual
```
conan install .. --build missing -o cppflow=on -pr:h=default -pr:b=default
conan build ..
```

File: .github/workflows/cmake.yml
Comment (tkoskela) [inline .github/workflows/cmake.yml]:
```suggestion
        version: 1.60.1
```

File: .github/workflows/documentation.yml
Comment (tkoskela) [inline .github/workflows/documentation.yml]: We should clean up the apt installed packages, similar to cmake.yml. Also install conan with the action

File: .github/workflows/linting.yml
Comment (tkoskela) [inline .github/workflows/linting.yml]:
```suggestion
        uses: actions/checkout@v3
```

File: cmake_files/LookUpSopt.cmake
Comment (tkoskela) [inline cmake_files/LookUpSopt.cmake:1]: Sopt gets installed by conan, so we should delete this file?

File: cmake_files/dependencies.cmake
Comment (tkoskela) [inline cmake_files/dependencies.cmake]: This was for debugging and can be deleted
```suggestion

```

File: cmake_files/dependencies.cmake
Comment (tkoskela) [inline cmake_files/dependencies.cmake:1]: I think all the `find_package` calls should have `REQUIRED` and we can drop the manual error handling

File: conanfile.py
Comment (tkoskela) [inline conanfile.py]:
```suggestion
        # Override version of zlib to prevent a conflict between versions of zlib required by depedencies
```

File: conanfile.py
Comment (tkoskela) [inline conanfile.py]:
```suggestion

```

File: conanfile.txt
Comment (tkoskela) [inline conanfile.txt:1]: conanfile.txt is not used in favour of conanfile.py, this should be deleted

File: cpp/tests/CMakeLists.txt
Comment (tkoskela) [inline cpp/tests/CMakeLists.txt:3]: This seems like the wrong way to check if we need spdlog, if that's what it's doing. Shouldn't this be `spdlog_FOUND` or `logging`?

Similar comments to `CATCH` and `sopt` below

File: cpp/tests/algo_factory.cc
Comment (tkoskela) [inline cpp/tests/algo_factory.cc:77]: This probably needs an explanation from #317

File: cpp/tests/algo_factory.cc
Comment (tkoskela) [inline cpp/tests/algo_factory.cc]: Doesn't really matter since we expect it to fail, but I would still remove the check `niters == 16`

File: cpp/tests/mpi_algo_factory.cc
Comment (tkoskela) [inline cpp/tests/mpi_algo_factory.cc]: I remember commenting these out because they were causing build errors, maybe worth checking if I can fix them :blush:

**Review by tkoskela — DISMISSED** [review]

File: .github/workflows/documentation.yml
Comment (tkoskela) [reply .github/workflows/documentation.yml]: It looks like the documentation workflow needs more updating. It's trying to build with `cmake` which will fail I assume.

**Comment by mmcleod89** [comment]
Looking at things here I think that this is up to date now with what I expect, keeping the forward-backward tensorflow test out of it for a separate PR. 👍

File: cpp/tests/CMakeLists.txt
Comment (tkoskela) [reply cpp/tests/CMakeLists.txt:3]: I don't really understand the logic of having these if statements, but I'm going to leave them alone for now. If the include dirs aren't found they don't get included in the target, but surely the build will then fail? Is it possible that they have been provided through some other way?

**Comment by tkoskela** [comment]
Ok, from my point of view the only outstanding item is [updating the documentation workflow](https://github.com/astro-informatics/purify/pull/302#discussion_r1261404711). I think everything else looks good to go.

File: README.md
Comment (SJaffa) [inline README.md:65]: This step created the `sopt` folder for me but didn't actually fetch the files, I had to add a `git submodule update --init --recursive` step. Is this the same for you @tkoskela @20DM ?

File: README.md
Comment (SJaffa) [inline README.md:71]: This gives an error about broken symlinks, which can be fixed by editing your `.conan/conan.conf` but did we find a way to pass that option on the command line?

File: README.md
Comment (SJaffa) [inline README.md]:
```suggestion
    mkdir build
    cd build
    conan install .. --build missing -o cppflow=on -pr:h=default -pr:b=default
```

Comment (SJaffa) [reply README.md]: Either need to be in the build folder or change the `..` to `.` in both the `conan install` and `conan build` commands

File: README.md
Comment (SJaffa) [inline README.md:148]: Should we remove this section since we have not maintained the Docker image? This isn't really related to this PR, just a general update so I'll make a separate issue

File: README.md
Comment (SJaffa) [inline README.md:131]: All tests are passing when building with or without `cppflow` but we don't have a test in Purify that checks `cppflow`, only is `SOPT`

File: README.md
Comment (tkoskela) [reply README.md:65]: I think this is because it does not recurse submodules in all branches, only the one it pulls when you clone. If I do
```
git clone --recurse-submodules --branch sj/conan-build https://github.com/astro-informatics/purify.git
```
I get all the submodules included, so I assume once this is merged into the default branch the current documentation will be fine.

File: README.md
Comment (tkoskela) [reply README.md:71]: Ah, I must have had that in my conf. That's a bit annoying. You could set an [environment varible](https://docs.conan.io/en/1.46/reference/env_vars.html#conan-skip-broken-symlinks-check)

File: README.md
Comment (tkoskela) [reply README.md:148]: My assumption was that if there is a self-contained dockerfile on dockerhub it might still work, hence the disclaimer I added below. I agree, create a separate issue and remove it from the documentation if the container is unusable.

File: README.md
Comment (tkoskela) [reply README.md:131]: This is addressed in #324

File: .github/workflows/documentation.yml
Comment (SJaffa) [reply .github/workflows/documentation.yml]: I've updated it to match the other workflows and copied the doxygen workaround we used in `sopt` but I can't test if it works as it only runs when we push to `development`.

File: README.md
Comment (SJaffa) [inline README.md]:
```suggestion
If you get an error about broken symlinks you can set `skip_broken_symlinks_check = True` in your `~/.conan/conan.conf` file or [set an environment variable](https://docs.conan.io/en/1.46/reference/env_vars.html#conan-skip-broken-symlinks-check)
```

File: README.md
Comment (SJaffa) [reply README.md:71]: I've added a note to the readme on how to fix this error.

File: README.md
Comment (SJaffa) [reply README.md:148]: I guess this isn't relevant to this PR so I'll make an issue and we can leave it for now, then I think this one is ready to merge!

**Comment by SJaffa** [comment]
I've checked the installation instructions in the readme work for me from a clean setup, I've updated the documentation workflow (although not tested this as it only runs when we push to development branch) and the other couple of outdated bits in the readme I've made new issues as they can be solved separately. So if someone could check over the docs workflow I think this is finally ready to merge! (assuming the tests pass...)

**Review by mmcleod89 — APPROVED** [review]
C++ code is good for me; as far as the docs action goes it looks okay to me but I don't think it's possible to run it manually so I'm happy for there to be a separate issue if there needs to be a further tweak there rather than extending this PR any further.

File: cmake_files/dependencies.cmake
Comment (mmcleod89) [reply cmake_files/dependencies.cmake:1]: I think we need to resolve this before it will let me merge; are we happy to move this to a different issue given that it is currently building?

Comment (20DM) [reply cmake_files/dependencies.cmake:1]: Happy to follow up on this in a separate merge request (will open an issue to remember).

**Pull request merged**
